feat(queue): allow to configure the train size (#2252)
Mergifyio/mergify-engine: docs/source/actions/queue.rst

The upside of creating multiple temporary pull requests is that continuous
integration pipelines can run on all of them in parallel. Currently, Mergify

> embarks up to 5 pull requests into the merge train (the ``speculative_checks``
> in ``queue_rules``).

Reviewer: ```suggestion
embarks up to 5 pull requests into the merge train. You can change this value with the``speculative_checks``
settings in ``queue_rules``.
```

     - list of ``conditions``
     -
     - The list of ``conditions`` to match to get the queued pull request merged.
   * - ``speculative_checks``
     - int
     - 5

>      - The number of car embarked to merge in the merge train.

Reviewer: As a user, if you're just skimming through the doc, there's 0 chance you understand car/train here.
```suggestion
     - The maximum number of checks to run in parallel in the queue.
```

I think it's still easy to understand. And if we ever have the batch-merge feature we can add

`merge_batch_size: maximum number of pull request to merge at the same time`

     - list of ``conditions``
     -
     - The list of ``conditions`` to match to get the queued pull request merged.
   * - ``speculative_checks``
     - int
     - 5
     - The maximum number of checks to run in parallel in the queue. Must be
       between 1 and 20.

>        See :ref:`speculative merges`

Reviewer: ```suggestion
       See :ref:`speculative merges`.
```
